include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js

```diff
@@ -0,0 +1,74 @@
+const CustomStrategy = require('passport-custom').Strategy;
+const strategyServices = require('./strategy_services');
+const request = require('request-promise');
+const Promise = require('bluebird');
+const SALESFORCE_OAUTH_TOKEN_URL = process.env.SALESFORCE_OAUTH_TOKEN_URL || '';
+const SALESFORCE_OAUTH_AUTHORIZE_URL = process.env.SALESFORCE_OAUTH_AUTHORIZE_URL || '';
+
+module.exports = (pb) => {
+    async function getSalesforceSettings(req) {
+        let pluginService = new pb.PluginService({
+            site: req.ctx.__site
+        });
+        pluginService = Promise.promisifyAll(pluginService);
+        const settings = await pluginService.getSettingsKVAsync('tn_join');
+        return settings;
```

Comment:
Since we are not in a try catch here, we could just return ```pluginService.getSettingsKvAsync('tn_join');```

right?  Not that I mind the way it is.

Reply:
I already add the try catch, I think you took a look to the old code :P